packages/go_router/lib/src/delegate.dart

```diff
@@ -116,10 +117,6 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
       final RouteBase route = match.route;
       if (route is GoRoute && route.parentNavigatorKey != null) {
         final bool canPop = route.parentNavigatorKey!.currentState!.canPop();
-        // Similar to popRoute, it should not be possible for a GoRoute with
-        // parentNavigatorKey to be the only page, so canPop should return true
-        // in this case.
-        assert(canPop);
```

Comment:
is this causing the issue?
also this can be simiply as 
```dart
assert(route.parentNavigatorKey!.currentState!.canPop());
return true;
```

Reply:
Yes, the assert is throwing when you pop from `/b` in the original issue (https://github.com/flutter/flutter/issues/111861)

Reply:
sounds good